src/main/java/com/github/teamfusion/spyglassplus/core/mixin/MouseMixin.java

```diff
@@ -32,18 +55,84 @@ public void onMouseScroll(long window, double horizontal, double vertical, Callb
 					}
 				}
 			}
-        }
-    }
+		}
+	}
 
-    @Override
-    public void addZoom(int zoom) {
-        this.zoom += zoom;
-    }
+	@Inject(method = "turnPlayer", at = @At("HEAD"), cancellable = true)
+	public void turnPlayer(CallbackInfo callbackInfo) {
+		double d0 = Blaze3D.getTime();
+		double d1 = d0 - this.lastMouseEventTime;
+		if (this.isMouseGrabbed() && this.minecraft.isWindowActive() && ((ISpyable) this.minecraft.player).getSpyGlassStands() != null) {
+			double d4 = this.minecraft.options.sensitivity * (double) 0.6F + (double) 0.2F;
+			double d5 = d4 * d4 * d4;
+			double d6 = d5 * 8.0D;
+			double d2;
+			double d3;
+			if (this.minecraft.options.getCameraType().isFirstPerson() && this.minecraft.player.isScoping()) {
+				this.smoothTurnX.reset();
+				this.smoothTurnY.reset();
+				d2 = this.accumulatedDX * d5;
+				d3 = this.accumulatedDY * d5;
+			} else {
+				this.smoothTurnX.reset();
+				this.smoothTurnY.reset();
+				d2 = this.accumulatedDX * d6;
+				d3 = this.accumulatedDY * d6;
+			}
 
-    @Override
-    public void setZero() {
-        this.zoom *= 0;
-    }
+			this.accumulatedDX = 0.0D;
+			this.accumulatedDY = 0.0D;
+			int i = 1;
+			if (this.minecraft.options.invertYMouse) {
+				i = -1;
+			}
+
+			this.minecraft.getTutorial().onMouse(d2, d3);
+			if (this.minecraft.player != null) {
+				turn((ISpyable) this.minecraft.player, d2, d3 * (double) i);
+			}
+			callbackInfo.cancel();
+		}
+	}
+
+	@Shadow
+	private boolean isMouseGrabbed() {
+		return false;
+	}
```

Comment:
As before, I don't believe this Shadow should have a method body unless I'm misunderstanding your intent.